plugins/events-correlation-engine/src/main/java/org/opensearch/plugin/correlation/core/index/query/CorrelationQueryBuilder.java

```diff
@@ -0,0 +1,334 @@
+    protected static final ParseField VECTOR_FIELD = new ParseField("vector");
+    protected static final ParseField K_FIELD = new ParseField("k");
+    protected static final ParseField FILTER_FIELD = new ParseField("filter");
+    /**
+     * max number of neighbors that can be retrieved.
+     */
+    public static int K_MAX = 10000;
+
+    /**
+     * name of the query
+     */
+    public static final ParseField NAME_FIELD = new ParseField("correlation");
+
+    private String fieldName;
+    private float[] vector;
+    private int k = 0;
+    private double boost;
+    private QueryBuilder filter;
+
+    private CorrelationQueryBuilder() {}
+
+    /**
+     * parameterized ctor for CorrelationQueryBuilder
+     * @param fieldName field name for query
+     * @param vector query vector
+     * @param k number of nearby neighbors
+     */
+    public CorrelationQueryBuilder(String fieldName, float[] vector, int k) {
+        this(fieldName, vector, k, null);
+    }
+
+    /**
+     * parameterized ctor for CorrelationQueryBuilder
+     * @param fieldName field name for query
+     * @param vector query vector
+     * @param k number of nearby neighbors
+     * @param filter optional filter query
+     */
+    public CorrelationQueryBuilder(String fieldName, float[] vector, int k, QueryBuilder filter) {
+        if (Strings.isNullOrEmpty(fieldName)) {
+            throw new IllegalArgumentException(
+                String.format(Locale.getDefault(), "[%s] requires fieldName", NAME_FIELD.getPreferredName())
+            );
+        }
+        if (vector == null) {
+            throw new IllegalArgumentException(
+                String.format(Locale.getDefault(), "[%s] requires query vector", NAME_FIELD.getPreferredName())
+            );
+        }
+        if (vector.length == 0) {
+            throw new IllegalArgumentException(
+                String.format(Locale.getDefault(), "[%s] query vector is empty", NAME_FIELD.getPreferredName())
+            );
+        }
+        if (k <= 0) {
+            throw new IllegalArgumentException(String.format(Locale.getDefault(), "[%s] requires k > 0", NAME_FIELD.getPreferredName()));
+        }
+        if (k > K_MAX) {
+            throw new IllegalArgumentException(String.format(Locale.getDefault(), "[%s] requires k <= ", K_MAX));
+        }
+
+        this.fieldName = fieldName;
+        this.vector = vector;
+        this.k = k;
+        this.filter = filter;
+    }
+
+    /**
+     * parameterized ctor for CorrelationQueryBuilder
+     * @param sin StreamInput
+     * @throws IOException IOException
+     */
+    public CorrelationQueryBuilder(StreamInput sin) throws IOException {
+        super(sin);
+        try {
+            this.fieldName = sin.readString();
+            this.vector = sin.readFloatArray();
+            this.k = sin.readInt();
+            this.filter = sin.readOptionalNamedWriteable(QueryBuilder.class);
+        } catch (IOException ex) {
```

Comment:
Why catch this as opposed to letting it propagate?

Reply:
fixed it.